### PR TITLE
feat(provider) add provider TrainingPeaks

### PR DESCRIPTION
**Pull request by squio**
# TrainingPeaks OAuth provider

## General

 - [x] Make sure you use [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
       Examples: `"fix(google): Fixed foobar bug"`, `"feat(accounts): Added foobar feature"`.
 - [x] All Python code must formatted using Black, and clean from pep8 and isort issues.
 - [x] JavaScript code should adhere to [StandardJS](https://standardjs.com).
 - [x] If your changes are significant, please update `ChangeLog.rst`.
 - [x] If your change is substantial, feel free to add yourself to `AUTHORS`.

 ## Provider Specifics

 In case you add a new provider:

- [x] Make sure unit tests are available.
- [x] Add an entry of your provider in `test_settings.py::INSTALLED_APPS` and `docs/installation.rst::INSTALLED_APPS`.
- [x] Add documentation to `docs/providers.rst`.
- [x] Add an entry to the list of supported providers over at `docs/overview.rst`.

### CODE REVIEW

File: allauth/socialaccount/providers/trainingpeaks/views.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        return "https://api." + self.get_hostname()
+
+    # https://oauth.sandbox.trainingpeaks.com/oauth/deauthorize
+
+
+    scope_delimiter = " "
+
+    def complete_login(self, request, app, token, **kwargs):
+        headers = {"Authorization": "Bearer {0}".format(token.token)}
+        response = requests.get(self.profile_url, headers=headers)
```
Comment (pennersr) [inline allauth/socialaccount/providers/trainingpeaks/views.py:52]: Can you please add `response.raise_for_status()` here?

Comment (squio) [reply allauth/socialaccount/providers/trainingpeaks/views.py:52]: Done, thanks!

File: allauth/socialaccount/providers/trainingpeaks/views.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+from allauth.socialaccount.providers.oauth2.views import (
+    OAuth2Adapter,
+    OAuth2CallbackView,
+    OAuth2LoginView,
+)
+
+from .provider import TrainingPeaksProvider
+
+
+class TrainingPeaksOauth2Adapter(OAuth2Adapter):
```
Comment (pennersr) [inline allauth/socialaccount/providers/trainingpeaks/views.py:13]: Can you please rename Oauth to OAuth ?

Comment (squio) [reply allauth/socialaccount/providers/trainingpeaks/views.py]: Nice one! Fixed and rebased branch with Master, should merge cleanly now.

Thanks!

**Pull request merged**